youtokentome/cpp/yttm.pyx

```diff
@@ -130,12 +132,30 @@ cdef class BPE:
             raise ValueError(status.message.decode())
         return subword.decode()
 
-    def decode(self, ids):
-        assert isinstance(ids, list)
+    def decode(self, ids, ignore_ids):
+
+        if not isinstance(ids, list):
+            raise TypeError(
+                "ids have to be a list instance but {} found".format(type(list))
+            )
+
+        if not (
+            isinstance(ignore_ids, Collection) or ignore_ids is None
+        ):
+            raise TypeError(
+                "ids have to be a Collection instance, but {} found".format(
+                    type(ignore_ids)
```

Comment:
`f"{type(ignore_ids)} is not a Collection type"` -- shorter.

Reply:
Fixed